hbase-oss/pom.xml

```diff
@@ -70,56 +71,206 @@
+                   you can ensure we don't gain additional transitive dependencies that need to
+                   be relocated. It can't be true when we build or the shade plugin will drop
+                   all of our direct dependencies at provided and test scope. :/
+                -->
+              <!--
+              <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
+              -->
+              <!-- XXX If you change what gets included, please review the generated
+                   LICENSE/NOTICE files to make sure they're accurate
+                -->
+              <artifactSet>
+                <includes>
+                  <include>com.google.*:*:jar:*</include>
+                  <include>org.apache.commons:commons-lang3:jar:*</include>
+                  <include>org.apache.curator:curator-*:jar:*</include>
+                  <include>org.apache.yetus:audience-annotations:jar:*</include>
+                  <include>org.apache.zookeeper:zookeeper:jar:*</include>
+                  <!-- transitives -->
+                  <include>io.netty:netty</include>
+                </includes>
+              </artifactSet>
+              <relocations>
+                <relocation>
+                  <pattern>com.google</pattern>
+                  <shadedPattern>${shading.prefix}.com.google</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.apache.commons</pattern>
+                  <shadedPattern>${shading.prefix}.org.apache.commons</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.apache.curator</pattern>
+                  <shadedPattern>${shading.prefix}.org.apache.curator</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.apache.jute</pattern>
+                  <shadedPattern>${shading.prefix}.org.apache.jute</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.apache.yetus</pattern>
+                  <shadedPattern>${shading.prefix}.org.apache.yetus</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.apache.zookeeper</pattern>
+                  <shadedPattern>${shading.prefix}.org.apache.zookeeper</shadedPattern>
+                </relocation>
+                <relocation>
+                  <pattern>org.codehaus</pattern>
+                  <shadedPattern>${shading.prefix}.org.codehaus</shadedPattern>
+                </relocation>
+                <!-- netty family -->
+                <relocation>
+                    <pattern>org.jboss.netty</pattern>
+                    <shadedPattern>${shading.prefix}.org.jboss.netty</shadedPattern>
+                </relocation>
+                <relocation>
+                    <pattern>io.netty</pattern>
+                    <shadedPattern>${shading.prefix}.io.netty</shadedPattern>
```

Comment:
Do we need to advise users about `io.netty.packagePrefix` and rename any of the shared objects from Netty?

I'm reminded of what we have in hbase-thirdparty: https://github.com/apache/hbase-thirdparty/blob/master/hbase-shaded-netty/pom.xml#L93-L137

Reply:
IIRC that's a Netty 4 issue and not present in Netty 3. The jar here has no JNI shared objects, fwiw.

Reply:
Ah! I thought `io.netty` was only introduced in Netty4, but I see I was mistaken. Agree with your assessment.